tensorboard/plugins/text/summary_test.py

```diff
@@ -0,0 +1,144 @@
+    return actual_proto
+
+  def normalize_summary_pb(self, pb):
+    """Pass `pb`'s `TensorProto` through a marshalling roundtrip.
+
+    `TensorProto`s can be equal in value even if they are not identical
+    in representation, because data can be stored in either the
+    `tensor_content` field or the `${dtype}_value` field. This
+    normalization ensures a canonical form, and should be used before
+    comparing two `Summary`s for equality.
+    """
+    result = tf.Summary()
+    result.MergeFrom(pb)
+    for value in result.value:
+      if value.HasField('tensor'):
+        new_tensor = tf.make_tensor_proto(tf.make_ndarray(value.tensor))
+        value.ClearField('tensor')
+        value.tensor.MergeFrom(new_tensor)
+    return result
+
+  def compute_and_check_summary_pb(self, name, data,
+                                   display_name=None, description=None,
+                                   data_tensor=None, feed_dict=None):
+    """Use both `op` and `pb` to get a summary, asserting equality.
+
+    Returns:
+      a `Summary` protocol buffer
+    """
+    if data_tensor is None:
+      data_tensor = tf.constant(data)
+    op = summary.op(
+        name, data, display_name=display_name, description=description)
+    pb = self.normalize_summary_pb(summary.pb(
+        name, data, display_name=display_name, description=description))
+    pb_via_op = self.normalize_summary_pb(
+        self.pb_via_op(op, feed_dict=feed_dict))
+    self.assertProtoEquals(pb, pb_via_op)
+    return pb
+
+  def test_metadata(self):
+    pb = self.compute_and_check_summary_pb('do', 'A deer. A female deer.')
+    summary_metadata = pb.value[0].metadata
+    plugin_data = summary_metadata.plugin_data
+    self.assertEqual(summary_metadata.display_name, 'do')
+    self.assertEqual(summary_metadata.summary_description, '')
+    self.assertEqual(plugin_data.plugin_name, metadata.PLUGIN_NAME)
+    content = summary_metadata.plugin_data.content
+    # There's no content, so successfully parsing is fine.
+    metadata.parse_plugin_metadata(content)
+
+  def test_explicit_display_name_and_description(self):
+    display_name = '"Re"'
+    description = 'A whole step above do.'
+    pb = self.compute_and_check_summary_pb('re', 'A drop of golden sun.',
+                                           display_name=display_name,
+                                           description=description)
+    summary_metadata = pb.value[0].metadata
+    self.assertEqual(summary_metadata.display_name, display_name)
+    self.assertEqual(summary_metadata.summary_description, description)
+    plugin_data = summary_metadata.plugin_data
+    self.assertEqual(plugin_data.plugin_name, metadata.PLUGIN_NAME)
+    content = summary_metadata.plugin_data.content
+    # There's no content, so successfully parsing is fine.
+    metadata.parse_plugin_metadata(content)
+
+  def test_string_value(self):
+    pb = self.compute_and_check_summary_pb('mi', 'A name I call myself.')
```

Comment:
This tests different things in Pythons 2 and 3. How about one test case for `b'A name I call myself'` and one for `u'A name I call myself'`?

It would be nice if the bytestring actually contained UTF-8 and the textstring actually contained Unicode, too.